api.go

```diff
@@ -560,6 +565,7 @@ func NewRaft(conf *Config, fsm FSM, logs LogStore, stable StableStore, snaps Sna
 		leaderNotifyCh:        make(chan struct{}, 1),
 		followerNotifyCh:      make(chan struct{}, 1),
 		mainThreadSaturation:  newSaturationMetric([]string{"raft", "thread", "main", "saturation"}, 1*time.Second),
+		preVoteDisabled:       conf.PreVoteDisabled || !transportSupportPreVote,
```

Comment:
perhaps we should also log a warning here? Something like `WARN: raft: pre-vote is disabled because it is not supported by the Transport`.